arelle/plugin/validate/EDINET/PluginValidationDataExtension.py

```diff
@@ -138,6 +145,33 @@ def _isDebitConcept(self, concept: ModelConcept) -> bool:
         """
         return bool(_DEBIT_QNAME_PATTERN.match(concept.qname.localName))
 
+    @lru_cache(1)
+    def checkElements(self, modelXbrl: ModelXbrl) -> ElementData:
+        instantAbstractElements: set[ModelConcept] = set()
+        instantDomainElements: set[ModelConcept] = set()
+        nonAbstractDomainElements: set[ModelConcept] = set()
+        nonNillableElements: set[ModelConcept] = set()
+        for modelDocument in modelXbrl.urlDocs.values():
+            for concept in modelDocument.xmlRootElement.iterdescendants(tag="{http://www.w3.org/2001/XMLSchema}element"):
+                if isinstance(concept,ModelConcept):
+                    if concept.namespaceURI == XbrlConst.xsd:
```

Comment:
Is there a reason not to do it this way? 
```suggestion
        for concept in modelXbrl.qnameConcepts.values():
```
As I alluded to in my other comment, if we've avoided the need to re-walk the XML tree for these rules, caching the results in `ElementData` doesn't seem to have a ton of benefit over just iterating over concepts separately one time for each rule (which would decouple these rules from each other).